genie-agent/src/main/java/com/netflix/genie/agent/AgentConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.agent;
+
+import com.netflix.genie.GenieAgentApplication;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * Configuration for shared agent beans.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@Slf4j
+class AgentConfig {
+
+    /**
+     * Setup bean validation.
+     *
+     * @return The bean validator
+     */
+    @Bean
+    @ConditionalOnMissingBean
+    public String hostName() {
```

Comment:
Maybe instead of creating beans of type `String` which is annoying and almost always needs qualifiers we should combine these two things into a Bean that is more of a properties container similar to how AWS exposes Instance metadata. That way it's a single bean of type `AgentInstanceMetadata` or something which we can pass around and modify as we go.

Reply:
Sounds good, should I do the same on the server?

Reply:
I think we probably should do something similar. I was looking at the Spring Cloud AWS earlier and they provide an [abstraction](http://cloud.spring.io/spring-cloud-static/spring-cloud-aws/2.0.0.M3/multi/multi__cloud_environment.html#_cloud_environment) on instance metadata that we could wrap with our own common interface. I don't think it should be the same interface on the server and agent if that's what you're asking